_ml-commons-plugin/algorithms.md

```diff
@@ -0,0 +1,220 @@
+        "column_type": "DOUBLE"
+      }
+    ],
+    "rows": [
+      {
+        "values": [
+          {
+            "column_type": "DOUBLE",
+            "value": 17.25701855310131
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+### Limitations
+
+ML Commons only supports the linear Stochastic gradient trainer or optimizer, which can not effectively map the non-linear relationships in trained data. When used with complicated data sets, the linear Stochastic trainer might cause some convergence problems and inaccurate results. 
+
+## RCF
+
+[Random Cut Forest](https://github.com/aws/random-cut-forest-by-aws) (RCF) is a probabilistic data structure used primarily for unsupervised anomaly detection. Its use also extends to density estimation and forecasting. OpenSearch leverages RCF for anomaly detection. ML Commons supports two new variants of RCF for different use cases:
+
+* Batch RCF: Detect anomalies in non-time-series data 
+* Fixed in time (FIT) RCF: Detect anomalies in time-series data
+
+### Parameters
+
+#### Batch RCF
+
+Parameter | Type   | Description | Default Value
+:--- |:--- | :--- | :---
+number_of_trees | integer | Number of trees in the forest | 30
+sample_size | integer | The same size used by the stream samplers in the forest | 256
+output_after | integer | The number of points required by stream samplers before results return | 32
+training_data_size | integer | The size of your training data | Data set size
+anamoly_score_threshold | double | The threshold of the anomaly score | 1.0 
+
+#### Fit RCF
+
+Parameter | Type   | Description | Default Value
+:--- |:--- | :--- | :---
+number_of_trees | integer | Number of trees in the forest | 30
+shingle_size | integer | A shingle, or consecutive sequence of the most recent records | 8
+sample_size | integer | The sample size used by stream samplers in the forest | 256
+output_after | integer | The number of points required by stream samplers before results return | 32
+time_decay | double | The decay factor used by stream samplers in the forest | 0.0001 
+anomaly_rate | double | The anomaly rate | 0.005
+time_field | string | (**Required**) The time filed for RCF to use as time-series data | N/A
+date_format | string | The date and time formatting for the time_field field | "yyyy-MM-ddHH:mm:ss"
+time_zone | string | The time zone for the time_field field | "UTC" 
+
+
+### APIs
+
+* [Train](https://opensearch.org/docs/latest/ml-commons-plugin/api/#train-model)
+* [Predict](https://opensearch.org/docs/latest/ml-commons-plugin/api/#predict)
+* [Train and predict](https://opensearch.org/docs/latest/ml-commons-plugin/api/#train-and-predict)
+
+
+### Limitations
+
+For FIT RCF, you can train the model with historical data, and store the trained model in your index. The model will be deserialized and predict new data points when using the Predict API. But the model in the index will not be refreshed with new data, because the model is "Fixed In Time".
```

Comment:
```suggestion
For FIT RCF, you can train the model with historical data, and store the trained model in your index. The model will be deserialized and predict new data points when using the Predict API. The model in the index will not be refreshed with new data, because the model is "Fixed In Time".
```